execute/factory.go

```diff
@@ -134,13 +134,16 @@ func (p PluginFactory) NewReportingPlugin(
 	// - Extended reader adds finality violation and contract binding management.
 	// - Observed reader adds metric reporting.
 	readers := make(map[cciptypes.ChainSelector]contractreader.ContractReaderFacade)
+	extended := make(map[cciptypes.ChainSelector]contractreader.Extended)
 	for chain, cr := range p.contractReaders {
 		chainID, err1 := sel.GetChainIDFromSelector(uint64(chain))
 		if err1 != nil {
 			return nil, ocr3types.ReportingPluginInfo{}, fmt.Errorf("failed to get chain id from selector: %w", err1)
 		}
-		readers[chain] = contractreader.NewExtendedContractReader(
+		reader := contractreader.NewExtendedContractReader(
 			contractreader.NewObserverReader(cr, lggr, chainID))
+		readers[chain] = reader
+		extended[chain] = reader
```

Comment:
We were always using the `Extended` reader, but it was being downcast to `ContractReaderFacade`. Another followup improvement would be to switch `NewCCIPChainReader` to `Extended` to stop double wrapping the downcast `Extended` object inside another `NewExtendedContractReader`.